hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/ConfigProviderImpl.java

```diff
@@ -48,7 +48,7 @@ public class ConfigProviderImpl implements ConfigProvider {
     public ConfigProviderImpl(@NonNull final PropertySource propertySource) {
         this.propertySource = Objects.requireNonNull(propertySource, "propertySource");
         final Configuration config = new ConfigurationAdaptor(propertySource);
-        configuration = new AtomicReference<>(new VersionedConfigImpl(config, 0));
+        this.configuration = new AtomicReference<>(new VersionedConfigImpl(config, 0));
```

Comment:
that ends in a sonar bug that says that volatile is wrong in that case...

Reply:
Interesting. Does it provide more details?

I think once I had a warning, because I used volatile on a class that was not thread-safe, which is weird. It's like saying, you must not use `final`, because the object you are referencing is mutable... I wonder, if it is the same issue here.

Reply:
created https://github.com/hashgraph/hedera-services/pull/6471 that does the change